.github/workflows/k3s-versions.yaml

```diff
@@ -0,0 +1,32 @@
+name: Update k3s-versions.json
+on:
+  schedule:
+  - cron: '43 8 * * *'
+  workflow_dispatch: {}
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  check-for-token:
+    outputs:
+      has-token: ${{ steps.calc.outputs.HAS_SECRET }}
+    runs-on: ubuntu-latest
+    steps:
+    - id: calc
+      run: echo "HAS_SECRET=${HAS_SECRET}" >> "${GITHUB_OUTPUT}"
+      env:
+        HAS_SECRET: ${{ secrets.RUN_WORKFLOW_FROM_WORKFLOW != '' }}
+
+  check-update-versions:
+    needs: check-for-token
+    if: needs.check-for-token.outputs.has-token == 'true'
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+      with:
+        fetch-depth: 0
```

Comment:
I don't think this is needed.

Reply:
It is needed now because we may add a commit to an existing branch.

Reply:
I thought you could push new commits to branches from a shallow clone, because the parent commit has a known commit hash? I'll test this in my fork, I guess.

Reply:
Yeah, this is not needed; since the parent commit hash doesn't change, having the default fetch depth of 1 is totally fine.